packages/cli/src/lib/push/requiredKeyValidator.ts

```diff
@@ -1,13 +1,13 @@
-import {SchemaValue} from '@coveo/bueno';
+import {PrimitivesValues, SchemaValue} from '@coveo/bueno';
 import {CaseInsensitiveDocument} from './caseInsensitiveDocument';
 
-export class RequiredKeyValidator<T> {
+export class RequiredKeyValidator<T extends PrimitivesValues> {
   private keys: string[];
-  private validations: {value: T; message: string | null}[] = [];
+  private validations: {value: T | unknown; message: string | null}[] = [];
```

Comment:
👍 as you said in the PR, lot of value are unknown. I think it's really nice to reflect that semantically thru the typings.